### ISSUE TITLE
Is there a setting for STDERR on console appenders?

### ISSUE THREAD

**Issue by ZLightning** [wood1986/ln#8]
I did not see any documented setting allowing users to specify if the console appender should log to STDOUT or STDERR.

**Comment by ZLightning** [comment]
For others looking for the same feature, switching console logger to STDERR can be accomplished in one line of code:

```javascript
var appenders = [{
   level: 'info',
   type: 'console',
   write: function (timestamp, string) { console.error(string.slice(0, -1)) },
}]
```

**Comment by wood1986** [comment]
No, currently there is no settings for STDERR. Actually, I do not expect people would use the "console". What I think is people will usually use the splunk or logstash to digest the log files. The console appender is just for development instead of production.

May I know your scenario?

**Comment by ZLightning** [comment]
ln is my go-to logger.  The specific need for STDERR is that some of my scripts use STDOUT for data output and the console logging conflicts with the data. My usage may be atypical as I rarely consume the logs into a database.  The data I want in elasticsearch I send directly from the script to an index.  I use the log files mostly for debugging and to confirm normal operation in production.

The programs I write generally involve text parsing of some sort and run on Ubuntu based servers.  It is often desirable for me to use bash to handle some portion of the output because my coworkers can modify or write the bash scripts.  The data gets redirected from STDOUT to a file or pipe, and the node script's progress can still be monitored on the console on STDERR.  It is much easier to shut off logging than to add it as an afterthought.

**Comment by wood1986** [comment]
give me 2 days

**Comment by wood1986** [comment]
Regardless of numbers of log files, why do not you write to two different files such as stdout.log and stderr.log? you can use bash or node script to watch them. Therefore, you do not have to redirect the output to a file or pipe. If you are doing that, why do you need ln?

**Comment by ZLightning** [comment]
Honestly, I don't need a fix to ln, as I already solved my problem.  I noted the work around in https://github.com/wood1986/ln/issues/8#issuecomment-234021330, and adding that to the documentation should be enough.

There are probably dozens of options for ways I could avoid using STDERR and STDOUT to send log messages and data to separate places.  My most recent script that I chose to move ln's console messages to STDERR was mostly about my time.  It allowed me to move on after the script accomplished the task it needed to do while a coworker put it to use.

I don't actually need ln in all the places I use it, but I do so because it is much easier when I reuse code.  I use ln through another module to expand the scope and expose the settings of loggers similar to winston, which I can't use because it is slow and introduced additional points of failure.  That module allows me to set the log level and appenders from the main module globally for any loggers.

In my experience, people do not always use programs I write the way I originally intended them to be used.  I love when my coworkers come up with creative uses for my programs.

Thank you for ln.  It is simple, fast and reliable.  If you want to see the code I use to manage the scope, I would be happy to provide it.
